### PR TITLE
Add function to create a framework dependent package dotnet-sdk containers

### PR DESCRIPTION
**Pull request by adityapatwardhan**
## PR Summary

Create a framework dependent package with reduced size specially for docker containers based on dotnet-sdk containers.

## PR Context  

The size of the framework dependent package is pretty large. It has assemblies of runtimes which are not required for the dotnet-sdk container images. This PR creates a new package with a reduced size.

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.  
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**  
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -2971,3 +2971,84 @@ function Get-PackageVersionAsMajorMinorBuildRevision
...
+    $fxdPackage = Get-ChildItem $FxdPackagePath -Recurse -Filter $basePackagePattern
+
+    Write-Verbose -Verbose "Fxd Package Path: $fxdPackage"
+
+    if ($fxdPackage) {
+        ## Extract fxd package
+        $tempExtractFolder = New-Item -Type Directory -Path "$FxdPackagePath/fxdreduced" -Force
+        Push-Location $tempExtractFolder
+
+        Write-Verbose -Verbose "Pushed location"
```
Comment (PaulHigin) [inline tools/packaging/packaging.psm1]: Seems like this would be more useful if the pushed location is included in the message.

**Review by PaulHigin — APPROVED** [review]

File: tools/packaging/packaging.psm1 (hunk 1 above)
Comment (adityapatwardhan) [reply tools/packaging/packaging.psm1]: Added. Thanks!

File: tools/packaging/packaging.psm1 [hunk 2]
```diff
@@ -2971,3 +2971,84 @@ function Get-PackageVersionAsMajorMinorBuildRevision
...
+        $packageNamePlatform = 'linux-x64'
+        $packageNameExtension = '.tar.gz'
+    }
+
+    $packageName = "powershell-$Version-$packageNamePlatform-fxd-dotnetsdk$packageNameExtension"
+
+    ## Get fxdependent package path
+    $fxdPackage = Get-ChildItem $FxdPackagePath -Recurse -Filter $basePackagePattern
+
+    Write-Verbose -Verbose "Fxd Package Path: $fxdPackage"
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1]: Can you use `Write-Log`?

File: tools/packaging/packaging.psm1 [hunk 3]
```diff
@@ -2971,3 +2971,84 @@ function Get-PackageVersionAsMajorMinorBuildRevision
...
+
+            Write-Verbose -Verbose "Compressing complete"
+
+        } finally {
+            Pop-Location
+        }
+    }
+
+    if (Test-Path "$FxdPackagePath/$packageName") {
+        Write-Host "##vso[artifact.upload containerfolder=release;artifactname=release]$FxdPackagePath/$packageName"
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1]: You are fine with this being just a message if we are not in Az DevOps?  I don't think it's a problem, just want to make sure it's not unintentional.

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
I would like you to fix the `Write-Log` comment before merging.

File: tools/packaging/packaging.psm1 [hunk 4]
```diff
@@ -2971,3 +2971,84 @@ function Get-PackageVersionAsMajorMinorBuildRevision
 
     $packageVersion
 }
+
+function New-DotnetSdkContainerFxdPackage {
```
[the remark this answers is not in the document]
Comment (adityapatwardhan) [reply tools/packaging/packaging.psm1]: Fixed.

File: tools/packaging/packaging.psm1 (hunk 3 above)
Comment (adityapatwardhan) [reply tools/packaging/packaging.psm1]: Yes, that should be ok.

File: tools/packaging/packaging.psm1 (hunk 2 above)
Comment (adityapatwardhan) [reply tools/packaging/packaging.psm1]: Fixed.

**Comment by adityapatwardhan** [comment]
@TravisEz13 can you re-review, your comments have been addressed.

**Comment by adityapatwardhan** [comment]
@anmenaga - Please have a look I have made changes to package verification tests.

**Review by anmenaga — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**